rir/src/compiler/native/builtins.cpp

```diff
@@ -71,41 +71,37 @@ SEXP createEnvironmentImpl(SEXP parent, SEXP arglist, int contextPos) {
     SLOWASSERT(TYPEOF(parent) == ENVSXP);
     SLOWASSERT(TYPEOF(arglist) == LISTSXP || arglist == R_NilValue);
     SEXP res = Rf_NewEnvironment(R_NilValue, arglist, parent);
+    PROTECT(res);
 
     if (contextPos > 0) {
         if (auto cptr = getFunctionContext(contextPos - 1)) {
             cptr->cloenv = res;
             if (cptr->promargs == symbol::delayedArglist) {
-                auto promargs = arglist;
-                bool hasMissing = false;
+                auto promargs = R_NilValue;
```

Comment:
it seems to me that previously we would share the promargs list with the environment in most cases, whereas now they are typically two lists. so this leads to massively more allocations, not?

Reply:
actually the `delayedArglist` stuff should probably be removed. see comment below